pkg/operator/clustermaintenancechange_v1.go

```diff
@@ -0,0 +1,436 @@
+			if strings.Contains(string(output), "not found") {
+				return false, nil
+			}
+
+			boolValue, err := parseStateOutput(output)
+			if err != nil {
+				return false, fmt.Errorf("error decoding is-managed attribute: %w", err)
+			}
+
+			// is-managed has the opposite logic than maintenance attribute
+			return !boolValue, nil
+		}
+	case nodeScope:
+		{
+			// this command fails if the node is unknown. Check the output to see if the node is recognized
+			// possible outputs:
+			// maintenance on: scope=nodes  name=maintenance value=true
+			// maintenance off: scope=nodes  name=maintenance value=off
+			// yes, it returns true/off instead of true/false, on/off...
+			// node not found output:
+			// Could not map name=node-name to a UUID
+			maintenanceMode, err := executor.Exec(ctx, "crm", "node", "attribute", args.nodeID, "show", "maintenance")
+			if err != nil && strings.Contains(string(maintenanceMode), "Could not map") {
+				return false, fmt.Errorf("error getting node maintenance attribute: %w", err)
+			}
+
+			values := nodeStatePatternCompiled.FindSubmatch(maintenanceMode)
+			if len(values) == 2 && string(values[1]) == "true" {
+				return true, nil
+			}
+
+			return false, nil
+		}
+	default:
+		{
+			maintenanceMode, err := executor.Exec(ctx, "crm", "configure", "get_property", "-t", "maintenance-mode")
+			if err != nil {
+				return false, fmt.Errorf("error getting maintenance-mode: %w", err)
+			}
+
+			boolValue, err := parseStateOutput(maintenanceMode)
+			if err != nil {
+				return false, fmt.Errorf("error decoding maintenance-mode attribute: %w", err)
+			}
+
+			return boolValue, nil
+		}
+	}
+}
+
+func setMaintenanceState(
+	ctx context.Context,
+	executor support.CmdExecutor,
+	scope clusterMaintenanceScope,
+	state bool,
+	args *clusterMaintenanceChangeArguments,
+) error {
+	strState := maintenanceOff
+	if state {
+		strState = maintenanceOn
+	}
+
+	switch scope {
+	case resourceScope:
+		{
+			_, err := executor.Exec(ctx, "crm", "maintenance", strState, args.resourceID)
+			return err
+		}
+	case nodeScope:
+		{
+			if state {
+				_, err := executor.Exec(ctx, "crm", "--force", "node", "maintenance", args.nodeID)
+				return err
+			}
+			_, err := executor.Exec(ctx, "crm", "--force", "node", "ready", args.nodeID)
+			return err
+		}
+	default:
+		{
+			_, err := executor.Exec(ctx, "crm", "maintenance", strState)
+			return err
+		}
+	}
+}
+
+func isIdle(ctx context.Context, executor support.CmdExecutor) error {
+	idleOutput, err := executor.Exec(ctx, "cs_clusterstate", "-i")
+	if err != nil {
+		return fmt.Errorf("error running cs_clusterstate: %w", err)
+	}
+
+	if !clusterIdlePatternCompiled.Match(idleOutput) {
+		return fmt.Errorf("cluster is not in S_IDLE state")
+	}
+
+	return nil
+}
+
+// depending on the queried resource, the crm command might print some "debug" lines
+// before returning the actual state of the attribute. That's why it needs some cleanup
```

Comment:
The actual state of the attribute is true or false right ? nothing else ? Maybe worth mentioning? 

Add maybe 
```
// Depending on the queried resource, the crm command might print debug lines before returning the actual 
// state of the attribute.
// The actual state is always a boolean value, either 'true' or 'false'
// The debug lines are cleaned up before parsing the final boolean state of the attribute.
```

This would help me :) what do you think ?

Reply:
I will update it